x/auth/vesting/types/period.go

```diff
@@ -26,3 +27,257 @@ func (vp Periods) String() string {
 	return strings.TrimSpace(fmt.Sprintf(`Vesting Periods:
 		%s`, strings.Join(periodsListString, ", ")))
 }
+
+// A schedule is an increasing step function of Coins over time.
+// It's specified as an absolute start time and a sequence of relative
+// periods, with each steps at the end of a period. A schedule may also
+// give the time and total value at the last step, which can speed
+// evaluation of the step function after the last step.
+
+// ReadSchedule returns the value of a schedule at readTime.
+func ReadSchedule(startTime, endTime int64, periods []Period, totalCoins sdk.Coins, readTime int64) sdk.Coins {
+	coins := sdk.NewCoins()
+
+	if readTime <= startTime {
+		return coins
+	}
+	if readTime >= endTime {
+		return totalCoins
+	}
+
+	time := startTime
+
+	for _, period := range periods {
+		x := readTime - time
+		if x < period.Length {
+			break
+		}
```

Comment:
I feel this could have better name + comment

Reply:
I had copied this from `vesting_account.go`, so I made minimal changes for ease of review. But I'm happy to refactor. Done.